cmd/kanikoExecute.go

```diff
@@ -101,6 +104,35 @@ func runKanikoExecute(config *kanikoExecuteOptions, telemetryData *telemetry.Cus
 		}
 	}
 
+	log.Entry().Infof("creating build settings information...")
+	stepName := "kanikoExecute"
+	dockerImage := ""
+	var dataParametersJSON map[string]interface{}
+	var errUnmarshal = json.Unmarshal([]byte(GeneralConfig.ParametersJSON), &dataParametersJSON)
+	if errUnmarshal != nil {
+		log.Entry().Infof("Reading ParametersJSON is failed")
+	}
+	if value, ok := dataParametersJSON["dockerImage"]; ok {
+		dockerImage = value.(string)
+	} else {
+		metadata, err := conf.ResolveMetadata(GeneralConfig.GitHubAccessTokens, GetAllStepMetadata, GeneralConfig.StepMetadata, stepName)
+		if err != nil {
+			log.Entry().Warnf("failed to resolve metadata: %v", err)
+		}
+		containers := metadata.Spec.Containers
+		if len(containers) > 0 {
+			dockerImage = containers[0].Image
+		}
+	}
+	kanikoConfig := buildsettings.BuildOptions{
+		DockerImage: dockerImage,
+	}
+	builSettings, err := buildsettings.CreateBuildSettingsInfo(&kanikoConfig, stepName)
```

Comment:
Choose another name, package `buildsettings` and variable `buildsettings` are too similar.

Reply:
Fixed, name was changed to buildSettingsInfo